### ISSUE TITLE
Trying to use FireOnChange<UserHeadOutputDevice> causes Flux tip to break and the node refuses to take a input at all

### ISSUE THREAD

**Issue by LeCloutPanda** [Yellow-Dog-Man/Resonite-Issues#1162]
### Describe the bug?

When trying to drag a UserHeadOutputDevice output into a FireOnChange<UserHeadOutputDevice> cause the tool to become "stuck" preventing me from using the tool and making me respawn the tool.

### To Reproduce

Use this layout of nodes and try connecting UserHeadOutputDevice to FireOnChange or vice versa 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/53411604/d2937dc5-86ee-4717-9d6a-4c5ce779a888)

### Expected behavior

It to connect properly.

### Screenshots

Look at the tool tip
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/53411604/2ed737ac-1703-4894-9f47-3756b3767870)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/53411604/ed4f3d85-d312-45e2-b7e2-878a717655e8)

### Resonite Version Number

2024.1.13.83

### What Platforms does this occur on?

Windows

### What headset if any do you use?

N/A

### Log Files

N/A

### Reporters

Discord and Resonite: LeCloutPanda

**Comment by LeCloutPanda** [comment]
Upon further digging it only does it for **FireOnObjectValueChange<UserHeadOutputDevice>** but using **FireOnTypeChange** it automatically changes it to the correct version and connects it up.

**Comment by Banane9** [comment]
Duplicate of #766 / #788 (which is itself a duplicate of #766, because it was closed before being reopened...).

TL;DR: You have to use the right version of FireOnChange, i.e. value or object, depending on what you want to connect. Otherwise it will throw an exception and get stuck, rather than switching type correctly.

**Comment by LeCloutPanda** [comment]
Ahh I see it was already an issue right my bad but one thing I did notice when I went to add a FireOnObjectValueChange<HeadOutputDevice> it is valid
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/53411604/d95a616b-623a-4619-8fe5-a37dab452ced)

and it is also valid when I do FireOnValueChange, guess I didn't try this method but it was weird that FireOnObjectValueChange let me put HeadOutputDevice
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/53411604/deb5f331-ebcd-4322-b3b6-7d2224a4b36c)

Anyways we can close this off because it's a duplicate of another.

**Comment by Banane9** [comment]
I'm not sure what exactly FireOnObjectValueChange is for, but it would seem to need a boxed value if anything.

You can close the issue too.
